src/textual/widgets/_data_table.py

```diff
@@ -223,36 +304,47 @@ class CellSelected(Message, bubble=True):
         Attributes:
             value: The value in the cell that was selected.
             coordinate: The coordinate of the cell that was selected.
+            cell_key: The key for the selected cell.
         """
 
         def __init__(
-            self, sender: DataTable, value: CellType, coordinate: Coordinate
+            self,
+            sender: DataTable,
+            value: CellType,
+            coordinate: Coordinate,
+            cell_key: CellKey,
         ) -> None:
             self.value: CellType = value
             self.coordinate: Coordinate = coordinate
+            self.cell_key: CellKey = cell_key
             super().__init__(sender)
 
         def __rich_repr__(self) -> rich.repr.Result:
             yield "sender", self.sender
             yield "value", self.value
             yield "coordinate", self.coordinate
+            yield "cell_key", self.cell_key
 
     class RowHighlighted(Message, bubble=True):
         """Posted when a row is highlighted. This message is only posted when the
-        `cursor_type` is set to `"row"`. Can be handled using `on_data_table_row_highlighted`
-        in a subclass of `DataTable` or in a parent widget in the DOM.
+        `cursor_type` is set to `"row"`. Can be handled using
+        `on_data_table_row_highlighted` in a subclass of `DataTable` or in a parent
+        widget in the DOM.
```

Comment:
May want reformatting to confirm with [our docstring guidelines](https://github.com/Textualize/textual/wiki/Documentation-standards#class).